projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -286,7 +286,7 @@ function get_element_styles_from_attributes( $attributes ) {
 
 		// Account for custom margins on inline forms.
 		$submit_button_styles .= true === get_attribute( $attributes, 'buttonOnNewLine' )
-			? sprintf( 'width: calc(100% - %spx;', get_attribute( $attributes, 'spacing', DEFAULT_SPACING_VALUE ) )
+			? sprintf( 'width: calc(100%% - %dpx);', get_attribute( $attributes, 'spacing', DEFAULT_SPACING_VALUE ) )
```

Comment:
Good catch. Should we actually have another case for buttons with 100% width on their own? By deducting the spacing from the width the subscribe button isn't quite full width which looks a little odd on my test site (on the front-end):

<img width="715" alt="Screen Shot 2022-12-22 at 2 49 53 pm" src="https://user-images.githubusercontent.com/16754605/209160814-ccd5dd4a-0ef6-46cd-aa17-5304d44b5ca7.png">

Something like this:

```
		if (  get_attribute( $attributes, 'buttonWidth' ) === '100%' ) {
			$submit_button_styles .= 'width: 100%;';
		}
		else {
			// Account for custom margins on inline forms.
			$submit_button_styles .= true === get_attribute( $attributes, 'buttonOnNewLine' )
				? sprintf( 'width: calc(100%% - %dpx);', get_attribute( $attributes, 'spacing', DEFAULT_SPACING_VALUE ) )
				: 'width: 100%;'; 
		}
```

I'm not too sure why we deduct spacing from the width anyway (it seems to me it should just be the space between the email field and button in this case) so I may be misunderstanding the original code.

What do you think about this?

Reply:
Actually that's probably better for a follow-up instead of here. I'll make sure this current fix gets in to the next release regardless.

Reply:
That's a good question. I'm not sure what the best fix is. Maybe we should be changing the default spacing value to 0?

We can see about fixing this in a follow-up for sure. 👍

Reply:
@coder-karen @jeherve Hi! where is this follow up happening? @obenland let me know that some of our patterns look broken because of the not quite 100% width button. 

For example,
* https://dotcompatterns.wordpress.com/2022/10/11/bold-newsletter-signup/
* https://dotcompatterns.wordpress.com/2022/10/11/newsletter-signup-with-media-on-the-left/

Reply:
@iamtakashi We didn't create any other PR as we weren't sure about the best approach to take. Do you have any opinion on the above?